### PR TITLE
Add more checks on repo status

### PR DESCRIPTION
**Pull request by AlexAxthelm**
Big additions here:

* Lists of "prod" vs "experimental" workflows/repos
* Function to check if a given repo is a dependency of a package on those lists (allows us to know if something is prod-critical or not)
* Get name of codeowner (from CODEOWNERS file) and if codeowners file has errors

### CODE REVIEW

**Comment by AlexAxthelm** [comment]
Work to continue:

- [ ] Check for workflows
- [ ] Check for rulesets
- [ ] include license in output

**Comment by jdhoffa** [comment]
What would a `CODEOWNERS` error look like?

File: R/repo_lists.R
Comment (jdhoffa) [inline R/repo_lists.R:23]: This file is missing all of the PROD transition monitor R packages: 
e.g.
``` r
transition_monitor_repos <- c(
  "RMI-PACTA/pacta.data.validation",
  "RMI-PACTA/pacta.scenario.data.preparation",
  "RMI-PACTA/pacta.data.scraping",
  "RMI-PACTA/pacta.data.preparation",
  "RMI-PACTA/pacta.portfolio.import",
  "RMI-PACTA/pacta.portfolio.audit",
  "RMI-PACTA/pacta.portfolio.allocate",
  "RMI-PACTA/pacta.executive.summary",
  "RMI-PACTA/pacta.portfolio.utils"
)
```

File: github-repos/github-repos.Rmd
Comment (jdhoffa) [inline github-repos/github-repos.Rmd:18]: I think you need to load the package here?
`library(pacta.sit.rep)`

File: R/repo_lists.R
Comment (AlexAxthelm) [reply R/repo_lists.R:23]: No, those are all dependencies of the workflows, so they'll show up as `prod` in the dependency tree

Comment (AlexAxthelm) [reply R/repo_lists.R:23]: More clearly, I explicitly did not include them here, since their "prod-critical" status is entirely defined by their role as a dependency

Comment (jdhoffa) [reply R/repo_lists.R:23]: Got it

**Comment by AlexAxthelm** [comment]
> What would a `CODEOWNERS` error look like?

The easiest way to get one is to misformat the CODEOWNERS file, but more generally laid out here: https://docs.github.com/en/rest/repos/repos?apiVersion=2022-11-28#list-codeowners-errors

File: github-repos/github-repos.Rmd
Comment (AlexAxthelm) [reply github-repos/github-repos.Rmd:18]: I've been using `devtools::load_all()` and rendering to html locally. 

Longer-term I think we would be better off putting any RMD templates in `inst/extdata`, and creating a `generate_sitrep` function that calls those templates with `system.file`, and that wshould handle all the library issues.

Comment (jdhoffa) [reply github-repos/github-repos.Rmd:18]: Happy to do that, but then this becomes a MUCH bigger project than I had initially envisioned for the `pacta.sit.rep` repo. (Not strictly against that, just a comment worth considering). 

The goal I had in mind for this repo was to be focused on communicating the state of our code to non-tech folks that are interested (e.g. Nayra, SLM, hodie, etc.). And also to say: hey here's a link to the most recently generated report. 

The functionality you suggest is super interesting, but definitely more developer focused. 

Maybe let's have a call to align on this (I don't think there's any reason that this repo can't support both use-cases, just want to make sure we are aligned)

**Comment by jdhoffa** [comment]
Hey @AlexAxthelm let me know if you want me to re-review?
This looks good but seems like you're still working on it

File: github-repos/github-repos.Rmd
Comment (AlexAxthelm) [reply github-repos/github-repos.Rmd:18]: Yeah, mostly I'm looking at the dev-focused stuff as related and naturally fitting here, rather than a separate repo, but I'm not worrying about making everythign "clean" for it yet.

**Comment by AlexAxthelm** [comment]
Yeah, still puttering on this for a little more. marking as draft. Added a check that the repo is using the standard `RMI-PACTA/actions` workflows.

TODO:
- [ ] check that expected workflows are expected (R.yml for R, docker.yml if a dockerfile is present)
- [ ] Rulesets

File: inst/extdata/github-repos.Rmd
Comment (jdhoffa) [inline inst/extdata/github-repos.Rmd:46]: When I try to run this, it errors out for me, weirdly on the `pacta.sit.rep` repo. 

Repro steps: 
* Source this `.Rmd`

When I ran it, it was the 62th repo that caused the failure: 
``` r
prod_checks(active_repos[[62]])
```

**Review by jdhoffa — CHANGES_REQUESTED** [review]
Syntactically it looks good, but I got a weird error when trying to source the `github-repos.Rmd` file, which I suppose is the main use-case of this repo? 

Not sure if there's some weirdness when it tries to fetch info on it's own repo?

File: inst/extdata/github-repos.Rmd
Comment (AlexAxthelm) [reply inst/extdata/github-repos.Rmd:46]: @jdhoffa That's because the CODEOWNERS file for this repo is in the wrong location (on `main`). I've added a fix in [abe689b](https://github.com/RMI-PACTA/pacta.sit.rep/pull/13/commits/abe689b43d2776810e5fdd400fa1f42535860830), but that won't resolve until it's been merged into `main`

Comment (jdhoffa) [reply inst/extdata/github-repos.Rmd:46]: Ahaha good catch

**Review by jdhoffa — APPROVED** [review]

**Pull request merged**
